From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Wed, 24 Jan 2024 21:40:17 +0800
Subject: [PATCH] =?UTF-8?q?C:=20dynamic=20Window=20title=20/=20=E5=8A=A8?=
 =?UTF-8?q?=E6=80=81=E6=B8=B8=E6=88=8F=E6=A0=87=E9=A2=98?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 core/src/mindustry/ClientLauncher.java | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/core/src/mindustry/ClientLauncher.java b/core/src/mindustry/ClientLauncher.java
index 1b5a25a7cbbd00c5a8e463ec722fb3b3ac5da9d6..46d9d838b42182024ebc8c0547361137900b1452 100644
--- a/core/src/mindustry/ClientLauncher.java
+++ b/core/src/mindustry/ClientLauncher.java
@@ -240,6 +240,8 @@ public abstract class ClientLauncher extends ApplicationCore implements Platform
                 }))));
             }
         }else{
+            updateTitle();
+
             asyncCore.begin();
 
             super.update();
@@ -256,6 +258,19 @@ public abstract class ClientLauncher extends ApplicationCore implements Platform
         }
     }
 
+    private String lastTitle = "";
+    private void updateTitle(){
+        if(graphics == null) return;
+        var mod = mods.orderedMods();
+        var title = "MindustryX | 版本号 " + Version.mdtXBuild +
+        " | mod启用" + mod.count(Mods.LoadedMod::enabled) + "/" + mod.size +
+        " | " + graphics.getWidth() + "x" + graphics.getHeight();
+        if(!title.equals(lastTitle)){
+            lastTitle = title;
+            graphics.setTitle(title);
+        }
+    }
+
     @Override
     public void exit(){
         //on graceful exit, finish the launch normally.
