From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Fri, 23 Feb 2024 01:05:54 +0800
Subject: [PATCH] API(Hooks) onHandleSendMessage

---
 core/src/mindustry/core/NetClient.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/src/mindustry/core/NetClient.java b/core/src/mindustry/core/NetClient.java
index 0948c4a880fe17643171857951722cd798248892..c6f8015a4616cb845c091c88425fd1cb33e0c8f4 100644
--- a/core/src/mindustry/core/NetClient.java
+++ b/core/src/mindustry/core/NetClient.java
@@ -197,6 +197,8 @@ public class NetClient implements ApplicationListener{
 
     @Remote(targets = Loc.server, variants = Variant.both)
     public static void sendMessage(String message, @Nullable String unformatted, @Nullable Player playersender){
+        message = mindustryX.Hooks.onHandleSendMessage(message, playersender);
+        if(message == null) return;
         if(Vars.ui != null){
             Vars.ui.chatfrag.addMessage(message);
             Sounds.chatMessage.play();
@@ -214,6 +216,8 @@ public class NetClient implements ApplicationListener{
     //equivalent to above method but there's no sender and no console log
     @Remote(called = Loc.server, targets = Loc.server)
     public static void sendMessage(String message){
+        message = mindustryX.Hooks.onHandleSendMessage(message, null);
+        if(message == null) return;
         if(Vars.ui != null){
             Vars.ui.chatfrag.addMessage(message);
             Sounds.chatMessage.play();
