From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sat, 2 Mar 2024 20:04:19 +0800
Subject: [PATCH] BUILD Kotlin 1.9.20

---
 build.gradle      | 11 ++---------
 gradle.properties |  2 +-
 2 files changed, 3 insertions(+), 10 deletions(-)

diff --git a/build.gradle b/build.gradle
index d64047c2ff2f1f6b4ead7354f4edee748c2347b1..e13f9df614deb5d24668053bf6e6ef0a417341bc 100644
--- a/build.gradle
+++ b/build.gradle
@@ -13,8 +13,8 @@ buildscript{
 }
 
 plugins{
-    id "org.jetbrains.kotlin.jvm" version "1.6.0"
-    id "org.jetbrains.kotlin.kapt" version "1.6.0"
+    id "org.jetbrains.kotlin.jvm" version "1.9.20"
+    id "org.jetbrains.kotlin.kapt" version "1.9.20"
 }
 
 allprojects{
@@ -180,13 +180,6 @@ project(":core"){
         }
     }
 
-    gradle.taskGraph.whenReady{
-        //these are completely unnecessary
-        tasks.kaptGenerateStubsKotlin.onlyIf{ false }
-        tasks.compileKotlin.onlyIf{ false }
-        tasks.inspectClassesForKotlinIC.onlyIf{ false }
-    }
-
     //comp** classes are only used for code generation
     jar{
         exclude("mindustry/entities/comp/**")
diff --git a/gradle.properties b/gradle.properties
index 5fcc2f6a7455a09fc17437e4f73e6e6f4d818f86..fad521726aa1c9205be065f31e31c75e845aeadd 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -18,7 +18,7 @@ kapt.use.worker.api=true
 # Compilation avoidance (see https://kotlinlang.org/docs/kapt.html#compile-avoidance-for-kapt-since-1-3-20)
 kapt.include.compile.classpath=false
 # I don't need to use the kotlin stdlib, so remove it to prevent extra bloat & method count issues
-kotlin.stdlib.default.dependency=false
+#kotlin.stdlib.default.dependency=false
 #needed for android compilation
 android.useAndroidX=true
 org.gradle.caching=true
