From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Thu, 4 Apr 2024 13:27:21 +0800
Subject: [PATCH] C: fix console messages order

---
 core/src/mindustry/Vars.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/src/mindustry/Vars.java b/core/src/mindustry/Vars.java
index ae716151e019fa61af85b99583d3bc8dba9a3e18..fde236399dcec5d7f5f96c101f8b6d398748c8fc 100644
--- a/core/src/mindustry/Vars.java
+++ b/core/src/mindustry/Vars.java
@@ -378,6 +378,10 @@ public class Vars implements Loadable{
                 if(!headless && (ui == null || ui.consolefrag == null)){
                     logBuffer.add(result);
                 }else if(!headless){
+                    if(!logBuffer.isEmpty()){
+                        logBuffer.each(ui.consolefrag::addMessage);
+                        logBuffer.clear();
+                    }
                     if(!OS.isWindows){
                         for(String code : ColorCodes.values){
                             result = result.replace(code, "");
