From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Fri, 19 Apr 2024 12:27:50 +0800
Subject: [PATCH] OC: no MapPreviewLoader.setupLoaders

---
 core/src/mindustry/ClientLauncher.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/src/mindustry/ClientLauncher.java b/core/src/mindustry/ClientLauncher.java
index f44dceb09dfef63bd11d3b54a75d0183976bb13c..177d7e2513ffaedc2104a982c14d1d0d45ae7e9a 100644
--- a/core/src/mindustry/ClientLauncher.java
+++ b/core/src/mindustry/ClientLauncher.java
@@ -143,7 +143,8 @@ public abstract class ClientLauncher extends ApplicationCore implements Platform
         assets.load("sprites/error.png", Texture.class);
         atlas = TextureAtlas.blankAtlas();
         Vars.net = new Net(platform.getNet());
-        MapPreviewLoader.setupLoaders();
+        //MDTX: optimize performance (inspired by Foo)
+//        MapPreviewLoader.setupLoaders();
         mods = new Mods();
         schematics = new Schematics();
 
