From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Fri, 19 Apr 2024 12:27:50 +0800
Subject: [PATCH] C: no MapPreviewLoader.setupLoaders

---
 core/src/mindustry/ClientLauncher.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/src/mindustry/ClientLauncher.java b/core/src/mindustry/ClientLauncher.java
index 91e7514c34ce5748998b14b98b799cf1bf572f0d..1a27c69e350fac21e9ff0ce4198f59b0e6304b9b 100644
--- a/core/src/mindustry/ClientLauncher.java
+++ b/core/src/mindustry/ClientLauncher.java
@@ -136,7 +136,8 @@ public abstract class ClientLauncher extends ApplicationCore implements Platform
         assets.load("sprites/error.png", Texture.class);
         atlas = TextureAtlas.blankAtlas();
         Vars.net = new Net(platform.getNet());
-        MapPreviewLoader.setupLoaders();
+        //MDTX: optimize performance (inspired by Foo)
+//        MapPreviewLoader.setupLoaders();
         mods = new Mods();
         schematics = new Schematics();
 
