From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sun, 15 Sep 2024 15:53:45 +0800
Subject: [PATCH] FC: reliableSync

---
 core/src/mindustry/net/Net.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/src/mindustry/net/Net.java b/core/src/mindustry/net/Net.java
index b6cffd4bf69edcd49f59794e615a4abc0bc16fdc..91d0ebaf667db45bf30862c92754407d9e812448 100644
--- a/core/src/mindustry/net/Net.java
+++ b/core/src/mindustry/net/Net.java
@@ -235,6 +235,7 @@ public class Net{
                 con.send(object, reliable);
             }
         }else{
+            reliable |= LogicExt.reliableSync;
             provider.sendClient(object, reliable);
         }
     }
