From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sun, 2 Jun 2024 13:10:51 +0800
Subject: [PATCH] OC: SDL_SetHint SDL_WINDOWS_DPI_SCALING

---
 desktop/src/mindustry/desktop/DesktopLauncher.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/desktop/src/mindustry/desktop/DesktopLauncher.java b/desktop/src/mindustry/desktop/DesktopLauncher.java
index 44f3b5b03f275b2d2151b9ef658879ddd5c06375..2d97d23f669f911849874b03e693edac3e85b0c0 100644
--- a/desktop/src/mindustry/desktop/DesktopLauncher.java
+++ b/desktop/src/mindustry/desktop/DesktopLauncher.java
@@ -70,6 +70,7 @@ public class DesktopLauncher extends ClientLauncher{
     }
 
     public DesktopLauncher(String[] args){
+        arc.backend.sdl.jni.SDL.SDL_SetHint("SDL_WINDOWS_DPI_SCALING", "1");
         Version.init();
         boolean useSteam = Version.modifier.contains("steam");
         testMobile = Seq.with(args).contains("-testMobile");
