From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sat, 25 May 2024 20:34:18 +0800
Subject: [PATCH] C(RenderExt) massDriverLine

---
 core/src/mindustryX/features/RenderExt.java | 26 +++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/core/src/mindustryX/features/RenderExt.java b/core/src/mindustryX/features/RenderExt.java
index 2dcf87814b002c0ef9349a02aafa6b01d1120400..398c77296e08680b6b1621f2f20174160c1acb92 100644
--- a/core/src/mindustryX/features/RenderExt.java
+++ b/core/src/mindustryX/features/RenderExt.java
@@ -4,6 +4,7 @@ import arc.*;
 import arc.graphics.*;
 import arc.graphics.g2d.*;
 import arc.math.*;
+import arc.math.geom.*;
 import arc.util.*;
 import mindustry.entities.*;
 import mindustry.game.EventType.*;
@@ -13,6 +14,7 @@ import mindustry.type.*;
 import mindustry.world.*;
 import mindustry.world.blocks.defense.*;
 import mindustry.world.blocks.defense.turrets.*;
+import mindustry.world.blocks.distribution.MassDriver.*;
 import mindustry.world.blocks.logic.*;
 import mindustry.world.blocks.logic.MessageBlock.*;
 import mindustry.world.blocks.production.Drill.*;
@@ -31,8 +33,11 @@ public class RenderExt{
     public static boolean logicDisplayNoBorder, arcDrillMode;
     public static int blockRenderLevel;
     public static boolean renderSort, renderMerge;
+    public static boolean massDriverLine;
+    public static int massDriverLineInterval;
 
     public static boolean unitHide = false;
+    public static Color massDriverLineColor = Color.clear;
 
     private static Effect placementEffect;
 
@@ -63,6 +68,8 @@ public class RenderExt{
             blockRenderLevel = Core.settings.getInt("blockRenderLevel");
             renderSort = Core.settings.getBool("renderSort");
             renderMerge = Core.settings.getBool("renderMerge");
+            massDriverLine = Core.settings.getBool("mass_driver_line");
+            massDriverLineInterval = Core.settings.getInt("mass_driver_line_interval");
         });
         Events.run(Trigger.draw, RenderExt::draw);
         Events.on(TileChangeEvent.class, RenderExt::onSetBlock);
@@ -84,6 +91,8 @@ public class RenderExt{
             Draw.draw(Layer.overlayUI - 0.1f, build::drawSelect);
         if(arcDrillMode && build instanceof DrillBuild drill)
             arcDrillModeDraw(block, drill);
+        if(massDriverLine && build instanceof MassDriverBuild b)
+            drawMassDriverLine(b);
     }
 
     private static void placementEffect(float x, float y, float lifetime, float range, Color color){
@@ -127,4 +136,21 @@ public class RenderExt{
             Lines.arc(dx, dy, iconSize * 0.75f, eff);
         }
     }
+
+    private static void drawMassDriverLine(MassDriverBuild build){
+        if(build.waitingShooters.isEmpty()) return;
+        Draw.z(Layer.effect);
+        float x = build.x, y = build.y, size = build.block.size;
+        float sin = Mathf.absin(Time.time, 6f, 1f);
+        for(var shooter : build.waitingShooters){
+            Lines.stroke(2f, Pal.placing);
+            Drawf.dashLine(RenderExt.massDriverLineColor, shooter.x, shooter.y, x, y);
+            int slice = Mathf.floorPositive(build.dst(shooter) / RenderExt.massDriverLineInterval);
+            Vec2 interval = Tmp.v1.set(build).sub(shooter).setLength(RenderExt.massDriverLineInterval);
+            float dx = interval.x, dy = interval.y;
+            for(int i = 0; i < slice; i++){
+                Drawf.arrow(shooter.x + dx * i, shooter.y + dy * i, x, y, size * tilesize + sin, 4f + sin, RenderExt.massDriverLineColor);
+            }
+        }
+    }
 }
