From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Wed, 24 Jul 2024 17:56:15 +0800
Subject: [PATCH] C: Portable Data Dir

---
 core/src/mindustry/ClientLauncher.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/src/mindustry/ClientLauncher.java b/core/src/mindustry/ClientLauncher.java
index 068a243ba87023201aaf0c287e9ee35341e312a3..519811a14b711d7192ad7d0b7d6e5b51e91b6b88 100644
--- a/core/src/mindustry/ClientLauncher.java
+++ b/core/src/mindustry/ClientLauncher.java
@@ -36,6 +36,9 @@ public abstract class ClientLauncher extends ApplicationCore implements Platform
     @Override
     public void setup(){
         String dataDir = OS.env("MINDUSTRY_DATA_DIR");
+        if(app.isDesktop() && dataDir == null && Fi.get("data").isDirectory()){
+            dataDir = "data";
+        }
         if(dataDir != null){
             Core.settings.setDataDirectory(files.absolute(dataDir));
         }
